### ISSUE TITLE
Question on add_training_cost

### ISSUE THREAD

**Issue by chao1224** [deepchem/deepchem#846]
In the `TensorflowGraphModel` , when calculating the loss [link](https://github.com/deepchem/deepchem/blob/2920bc29185378966583fcf38793b2bb276547ba/contrib/atomicconv/models/legacy.py#L282)
Should it be ` tf.div(tf.reduce_sum(weighted_cost), tf.reduce_sum(weights[task]))`? Since there can be missing values in each batch.

**Comment by rbharath** [comment]
This is a reasonable proposal, but might lead to problems if we have batches with zero weight (which does occasionally happen due to random shuffles). I think the current code is more stable and avoids this error. My gut feeling is it shouldn't change perf too much in any case.

**Comment by chao1224** [comment]
@rbharath Thanks. Now I'm using PyTorch but cannot reach the baseline performance, so I just try to find any difference that can lead to such difference.

For zero weight, I simply set it as zero, which has no effect to the gradient updates in PyTorch, not sure how TF handles it.

And your instince is right, to test this, I tried both averaged method, the difference is not too big.
